presto-testing-docker/src/main/java/io/prestosql/testing/TemporaryDockerContainer.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.testing;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.spotify.docker.client.DefaultDockerClient;
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.messages.Container;
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.HostConfig;
+import com.spotify.docker.client.messages.PortBinding;
+import io.airlift.log.Logger;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import net.jodah.failsafe.function.CheckedRunnable;
+
+import java.io.Closeable;
+import java.net.Socket;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static com.google.common.collect.MoreCollectors.toOptional;
+import static java.lang.String.format;
+import static java.time.temporal.ChronoUnit.MINUTES;
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static java.util.Objects.requireNonNull;
+
+public class TemporaryDockerContainer
+        implements Closeable
+{
+    private static final Logger LOG = Logger.get(TemporaryDockerContainer.class);
+
+    private final String hostIp = "127.0.0.1";
+    private final String image;
+    private final String hostName;
+    private final Map<String, String> environment;
+    private DockerClient dockerClient;
+    private String containerId;
+    private Map<Integer, Integer> hostPorts;
+
+    public TemporaryDockerContainer(String image, String hostName, List<Integer> ports, Map<String, String> environment)
+    {
+        this.image = requireNonNull(image, "image is null");
+        this.hostName = requireNonNull(hostName, "hostName is null");
+        this.environment = ImmutableMap.copyOf(requireNonNull(environment, "environment is null"));
+        try {
+            setUp(ports);
```

Comment:
This looks too much like "set up ports". I misunderstood this twice already
please assign `ports` to a field

Reply:
I think with proper fields names (e.g as below) this will be clear.
```
private final List<Integer> ports; // or openPorts
private Map<Integer, Integer> hostPorts;
```